src/internal/hooks/use-visual-mode/index.ts

```diff
@@ -5,6 +5,11 @@ import { ALWAYS_VISUAL_REFRESH } from '../../environment';
 import { isDevelopment } from '../../is-development';
 import { warnOnce } from '@cloudscape-design/component-toolkit/internal';
 
+interface ExtendedWindow extends Window {
+  [key: symbol]: (() => boolean) | undefined;
```

Comment:
I think you can get more type safety by declaring the symbol via

```js
const awsuiVisualRefreshFlag = Symbol.for('awsui-visual-refresh-flag');
```

Check [this code demo](https://www.typescriptlang.org/play?ssl=3&ssc=1&pln=3&pc=72#code/KYDwDg9gTgLgBAbwL4G4BQaDGEB2BneAQwHc8BXASwDULzCAbAJWADMpg8ALAMXsIHM4AXjgBlAJ4BbAEYR6AOhbQAFAHIS5CgFoAbrTIMt7Nh05aWffqoCU6NBRwxgUFoUzA4AURBOcAE2A-AHUHPwhiOFBfPzw4EP9wxDQ4OABtDUoaOiZWdi5eAQBdAH4ALjhla2EAPjhZOWBCHHQkDDQAzD52OGx8eGJQ8PLvaMD4sOI7NAB6abhiaABrPCxcPAb5egh+ZQGE4nTSTP0GZhN8yxL5StsMWbgIGE5nODwpWXpYsLgcR-ml1b4DZbHZ7CapCQyOSKFSqdaSYBPBz8LTAT7AGyFaxAA) for details